app/views/devise/sessions/two_factor.html.haml

```diff
@@ -1,7 +1,20 @@
+:ruby
+  browser = Browser.new(request.user_agent)
+  device_name = format("%s %s", browser.name, browser.platform.name)
+  button_label = format("Remember this device for %s", TwoFactorsToken.active_duration.inspect)
+
 - @showing_sign_in_form = true
 %h1 Two-Factor Authentication
 .row
   .col-sm-4
     = simple_form_for(resource, as: resource_name, url: session_path(resource_name), method: :post, wrapper: :vertical_form_hint_first) do |f|
       = f.input :otp_attempt, label: 'Two-Factor Authentication code', required: true, autofocus: true
-      = f.button :submit, "Verify code"
\ No newline at end of file
+      = f.input :remember_device, as: :boolean, label: button_label, checked_value: true, unchecked_value: false, input_html: { checked:false, class: :jRememberDevice }
+      = f.input :device_name, input_html: { value: device_name }, wrapper_html: { class: :jDeviceName, style: 'display: none' }
```

Comment:
I think you can get the device name directly from `browser.device.name` , we don't need to let the user set that manually, we'll just use something we can fetch automatically..

Reply:
I have it set up so that the default name is set automatically, so the user could choose to rename the device on this step if they want to, in cases where they have multiple devices with the same default name for example.

Otherwise you could just click continue like usual